resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/ResourceRequestHelper.scala

```diff
@@ -153,7 +153,14 @@ private object ResourceRequestHelper extends Logging {
 
     val resInfoClass = Utils.classForName(RESOURCE_INFO_CLASS)
     val setResourceInformationMethod =
-      resource.getClass.getMethod("setResourceInformation", classOf[String], resInfoClass)
+      try { 
+        resource.getClass.getMethod("setResourceInformation", classOf[String], resInfoClass)
+      } catch {
+        case e: NoSuchMethodException => 
+          logError(s"Cannot find $RESOURCE_INFO_CLASS.setResourceInformation. This is likely due to a jar conflict between different yarn versions. Unable to set resources.")
```

Comment:
Hi, @abellina .
Thank you for your contribution. Before making a PR, you can check the scala style rule. For example, this line will fails at Scala Style check. Could you fix it?
```
$ dev/scalastyle
```
```

Reply:
@dongjoon-hyun sorry about that. Fixed.